### PR TITLE
Import wiki content about Smarty templating

### PR DESCRIPTION
**Pull request by seanmadsen**
This PR imports the following wiki pages: 

* https://wiki.civicrm.org/confluence/display/CRMDOC/Page+Templates
* https://wiki.civicrm.org/confluence/display/CRMDOC/Customize+Built-in+Profile+Contribution+and+Event+Registration+Screens
* https://wiki.civicrm.org/confluence/display/CRMDOC/Extending+Smarty

**I would like someone else to review** this before merging. Maybe you even want make some improvements to this PR yourself? That would be great because I'm likely to agree with anything you suggest. Also I'm fine with merging as-is.

Notes: 

* I think the level of quality for this content is a bit sub-par, but I don't feel confident enough in the subject matter to improve it myself. 
* For the most part, I kept my edits to formatting-only.
* I put them all in one place, but I'm not sure this is the best way to organize this content.

### CODE REVIEW

**Comment by seamuslee001** [comment]
I have read over this and i believe it makes sense and looks good to me.  Agree its historical but still need to have it in

**Comment by seanmadsen** [comment]
Thanks @seamuslee001

File: docs/framework/templates/customizing.md [hunk 1]
```diff
@@ -0,0 +1,411 @@
+# Customizing Built-in Screens
+
+!!! caution "Use extensions instead"
+
+    The methods on this page are of historical interest and may be helpful in understanding legacy code, but overriding templates and php files is strongly discouraged for any new customisations. These approaches need attention whenever you upgrade and will cause a range of problems from subtle to major if you forget about them.
+    
+    As CiviCRM evolves, the recommended ways of customizing it change.   The current preferred method is to Create an Extension to implement Hooks and use the API.  If you can't find a way to use existing hooks to accomplish what you want, discuss it on StackExchange or Chat.  It may be appropriate to add a new hook to core.
+    
+    (It is possible to override templates and php in an extension, but you face the same problems when it comes to upgrading - don't do it!)
```
Comment (mattwire) [inline docs/framework/templates/customizing.md]: Useful extension to help with this is: https://civicrm.org/extensions/extension-file-overrides

File: docs/framework/templates/index.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+If your modifications go further than "simple" modifications of the
+layout, but need to display more content than the one assigned to the
+template by default, or to add Ajax functionality, use the CiviCRM API.
+Please read more information about using the CiviCRM API from Ajax to
+pursue this approach.
+
+In most cases, using the CiviCRM APIs should be simple and only takes a
+few extra lines of modifications.
+
+## Appending jQuery or other code to a template
```
Comment (mattwire) [inline docs/framework/templates/index.md]: Better to use hooks and `CRM_Core_Resources::singleton()->addScriptFile` here.  Do we have a part of the docs that explains this?

**Comment by seanmadsen** [comment]
@mattwire would you be willing to make a new pull request with your suggested changes?

**Pull request merged**